README.md

```diff
@@ -272,6 +272,27 @@ This is a general procedure on how to do it:
 
 Both of these monitoring base classes are dashboard segments, so you can add them to your monitoring by calling `.addSegment()`.
 
+### Custom dashboards
+
+If you want *even* more flexibility, you can take complete control over dashboard generation by leveraging dynamic dashboarding features. This allows you to create an arbitrary number of dashboards while configuring each of them separately. Here's how:
+
+1. Create an instance of your monitoring facade by instantiating a `DefaultDynamicDashboardFactory` and providing it via props.
+2. This will require providing an array of `DynamicDashboardConfiguration` objects, which will require at least specifying names for dashboards you want generated. For example, if you wanted to create two dashboards, one for your AWS Hosted Service metrics and one for AWS Supporting Infrastructure Metrics, you would do this:
+    ```
+    const dynamicDashboardFactory = new DefaultDynamicDashboardFactory(
+      stack,
+      "TestDynamicDashboardFactory",
+      {
+          dashboardNamePrefix: "GenericHostedAWSService",
+          dashboardConfigs: [{ name: "HostedService" }, { name: "Infrastructure" }],
+      }
+    );
+    ````
+    This would generate two dashboards with the names "GenericHostedAWSService-HostedService" and "GenericHostedAWSService-Infrastructure"
+3. Identify the monitoring constructs used by your application and augment them by implementing the `IDynamicDashboardSegment` interface.
+4. Replace `monitoring.addSegment` calls in your application with `monitoring.addDynamicSegment` calls and provide the extended implementations you created in the previous step.
```

Comment:
I'd probably just make this a fuller code example of a facade with the dashboard factory defined with supplementary comments explaining the relevant bits.

Likewise for step 5, which an example of something with a `widgetsByDashboardType` method.

Reply:
I will save the README updates for after we have aligned on the coding contract to avoid unnecessary iterations. let me know if this is OK.

Reply:
Yes, that's fine.